src/web/mjs/connectors/manmanapp.mjs

```diff
@@ -0,0 +1,79 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class ManmanApp extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'manmanapp';
+        super.label = 'Manman Comic 漫漫漫画';
+        this.tags = [ 'manga', 'webtoon', 'chineses' ];
+        this.url = 'https://www.manmanapp.com';
+    }
+
+    async _getMangaFromURI(uri) {
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, 'li.title');
+        console.log(data)
+        let id = uri.pathname + uri.search;
+        let title = data[0].firstChild.textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        let request = new Request(new URL('/comic/category_1.html', this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'div.paging li:last-of-type > a');
+        let pageCount = parseInt(data[0].href.match(/(\d+).html/)[1]);
+        for(let page = 1; page <= pageCount; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        let request = new Request(new URL(`/comic/category_${page}.html`, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'li.title > a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let chaptersList = [];
+        const id = manga.id.match(/(\d+).html/)[1]
+        for(let page = 1, run = true; run; page++) {
+            let chapters = await this._getChaptersFromPage(page,id);
+            chapters.length > 0 ? chaptersList.push(...chapters) : run = false;
+        }
+        return chaptersList;
+    }
+
+    async _getChaptersFromPage(page,id) {
+        let request = new Request(new URL('/works/comic-list-ajax.html', this.url), {
+            method: 'POST',
+            body: new URLSearchParams({
+                id: id,
+                sort: 0,
+                page: page
+            })
```

Comment:
Does this work without `.toString()`?
Does this work without setting `content-type` header to `application/x-www-form-urlencoded`?

Reply:
they are numbers why would i have to `.toString()`
and yes it works without setting `content-type` header to `application/x-www-form-urlencoded`

Reply:
its `content-type: application/x-www-form-urlencoded;charset=UTF-8` without setting it

Reply:
`URLSearchParams` is an object, but body is expected to use a string, thats why usually `URLSearchParams` is required to being serialized with `URLSearchParams.toString`